File: algorithm/src/main/java/baekjoon/p02523/Main.java
package baekjoon.p02523;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt()));
    }

    private static String solve(int n) {
        StringBuilder sb = new StringBuilder();

        for (int i = 1; i <= n; i++) {
            sb.append(repeat("*", i));
            sb.append("\n");
        }

        for (int i = n - 1; i >= 1; i--) {
            sb.append(repeat("*", i));
            sb.append("\n");
        }

        return sb.toString();
    }

    private static String repeat(String str, int cnt) {
        StringBuilder sb = new StringBuilder();

        while (cnt-- > 0) {
            sb.append(str);
        }

        return sb.toString();
    }

}

File: algorithm/src/main/java/baekjoon/p02523/Main.kt
package baekjoon.p02523

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    println(solve(n))
}

private fun solve(n: Int): String {
    return buildString {
        for (i in 1..n) {
            append("*".repeat(i))
            append("\n")
        }

        for (i in n - 1 downTo 1) {
            append("*".repeat(i))
            append("\n")
        }
    }
}